### PR TITLE
feat: remove disallowed sentry event properties

### PR DESCRIPTION
**Pull request by kajm**
## Explain the changes you’ve made
Remove event properties that is considered as personal information

## Explain why these changes are made
We don't need user specific data when troubleshooting, hence removing it.

## Explain your solution
Specify an array of paths in the Sentry.Event object where user data is. Use lodash set to set a new undefined value on that specific path.

1. Checkout this branch
2. Fire upp the simulator by running the command `yarn ios` or `yarn android`
3. Make the array send data to sentry by throw in app.tsx as an example

## Tested environments

- [] Storybook on a iOS device/simulator.
- [x] Building the Application on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-3hdexwf Ladda inte upp information till Sentry som inte är nödvändig för felsökning](https://app.clickup.com/t/3hdexwf)

**Review by MasterKenth — CHANGES_REQUESTED** [review]
Does not work as intended. beforeSend doesn't seem to recieve an event with device data, so after scrubbing it goes from `"contexts": {}` to `"contexts": { "device": {} }` which I suspect overrides _all_ device info afterwards, since the device tab for the event in Sentry now is completely empty.

File: source/services/monitoring/SentryMonitoringService.ts [hunk 1]
```diff
@@ -1,15 +1,33 @@
...
     Sentry.init({
       dsn: env.SENTRY_DSN,
       // Set tracesSampleRate to 1.0 to capture 100% of transactions for performance monitoring.
       // Recommended is to adjust this value in production.
       tracesSampleRate: 0.75,
+      beforeSend: (event) => {
+        let eventCopy = deepCopy(event);
+
+        forbiddenSentryEventProperties.forEach((path: string) => {
+          eventCopy = setObjectPathValue(eventCopy, path, undefined);
```
Comment (MasterKenth) [inline source/services/monitoring/SentryMonitoringService.ts:26]: Potentially move to using a specific "unset" function instead.

File: source/services/monitoring/SentryMonitoringService.ts [hunk 2]
```diff
@@ -1,15 +1,33 @@
...
     Sentry.init({
       dsn: env.SENTRY_DSN,
       // Set tracesSampleRate to 1.0 to capture 100% of transactions for performance monitoring.
       // Recommended is to adjust this value in production.
       tracesSampleRate: 0.75,
+      beforeSend: (event) => {
+        let filteredEvent: Sentry.Event = {};
+
+        forbiddenSentryEventProperties.forEach((path: string) => {
+          filteredEvent = unsetObjectPathValue<Sentry.Event>(event, path);
```
Comment (MasterKenth) [inline source/services/monitoring/SentryMonitoringService.ts:26]: Template parameter `<Sentry.Entry>` is superfluous - type can be inferred.

**Review by MasterKenth — CHANGES_REQUESTED** [review]

**Review by MasterKenth — APPROVED** [review]

File: source/services/monitoring/SentryMonitoringService.ts [hunk 3]
```diff
@@ -1,15 +1,33 @@
 import * as Sentry from "@sentry/react-native";
 import env from "react-native-config";
 
+import { unsetObjectPathValue } from "../../helpers/Objects";
+
 import type { MonitoringService } from "./MonitoringService.types";
 
+const forbiddenSentryEventProperties = [
```
Comment (dannil76) [inline source/services/monitoring/SentryMonitoringService.ts:8]: Är det kanske lättare att tänka strikt så att vi väljer de vi VILL tillåta. Kanske kan bli jobbigt att veta på förhand alla som vi inte vill ha med?

**Review by dannil76 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**